### DISCUSSION TITLE
Not able to solve IDE using the modied code from Voltera_IDE.py

### DISCUSSION THREAD

**Discussion by JunaidAkhter** [lululxvi/deepxde#1127] [Q&A]
Hi,

I am trying to solve the following integral differential equation:

du(x)/dx = 1 - int_{0}^{x} u(t) dt with the initial condition u(0) = 0. The exact solution is : u(x) = sin(x)

I used the following code to get the solution. 

import deepxde as dde
import matplotlib.pyplot as plt
import numpy as np
from deepxde.backend import tf

def ide(x, y, int_mat):
    lhs1 = tf.matmul(int_mat, y)
    rhs = tf.gradients(y, x)[0]
    return (rhs-1)[: tf.size(lhs1)] + lhs1

**WE DO NOT HAVE KERNEL FOR THIS EXAMPLE **
""" def kernel(x, s):
    return np.exp(s - x)
 """ 

def func(x):
    """
    x: array_like, N x D_in
    y: array_like, N x D_out
    """
    return np.sin(2 * np.pi * x)

geom = dde.geometry.TimeDomain(0, 1)
ic = dde.icbc.IC(geom, func, lambda _, on_initial: on_initial)

quad_deg = 20
data = dde.data.IDE(
    geom,
    ide,
    ic,
    quad_deg,
    num_domain=16,
    num_boundary=2
)

layer_size = [1] + [20] * 3 + [1]
activation = "tanh"
initializer = "Glorot uniform"
net = dde.nn.FNN(layer_size, activation, initializer)

model = dde.Model(data, net)

#model.compile("L-BFGS")
#model.train()
*** THE ABOVE TWO LINES DO NOT WORK ALTHOUGH THEY ARE MENTIONED IN THE ORIGINAL CODE***

** THE FOLLOWING TWO LINES WORK **
model.compile("adam", lr=0.001)
model.train(iterations=10000)

X = geom.uniform_points(100)
y_true = func(X)
y_pred = model.predict(X)
print("L2 relative error:", dde.metrics.l2_relative_error(y_true, y_pred))

plt.figure()
plt.plot(X, y_true, "-")
plt.plot(X, y_pred, "o")
plt.show()
np.savetxt("test.dat", np.hstack((X, y_true, y_pred)))

However, the solution which I get after training is completely wrong. I tried to solve some other Integral Differential Equation but I was not able solve them do. I think I am doing something trivially wrong as there  is not a lot of modification that I need to do the original code to be able to solve this equation. It would be great if you could point out what could be the issue. 
![Voltera](https://user-images.githubusercontent.com/51301551/215809430-c456b30a-5aa0-403b-ac5d-f702e9120e4f.png)

**Comment by lululxvi** [comment]
Your code is hard to read. Please format the code.
